### DISCUSSION TITLE
Call to undefined function Google\Protobuf\Internal\bccomp()

### DISCUSSION THREAD

**Discussion by ghost** [freekmurze/laravel-analytics#510] [General]
Everything is working fine in local server but on `production` and `staging` server. I am getting this error.

```php
Call to undefined function Google\Protobuf\Internal\bccomp()
```

```php
use Spatie\Analytics\Facades\Analytics;
use Spatie\Analytics\Period;

$period = match ($this->interval) {
    'last-7-days' => Period::days(7),
    'last-30-days' => Period::days(30),
    'last-6-months' => Period::months(6),
    'last-1-year' => Period::years(1),
};

$analyticsData = Analytics::fetchTopCountries($period);
```

Thank you for the great package.

**Comment by ghost** [comment]
I tried installing `protobuf` and `grpc` php extension. Now I'm getting `Bad Gateway: 502` error.

**Comment by xuanvu2911** [comment]
"require": {
        "php": "^8.1",
        "ext-gd": "*",
        "ext-bcmath": "*",
....
You can add ext via composer.json , then run composer update and commit it
